vm/package/execute.rs

```diff
@@ -0,0 +1,342 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+use snarkvm_compiler::Transaction;
+use snarkvm_console::{
+    program::{Plaintext, Record},
+    types::Address,
+};
+
+use super::*;
+
+pub struct ExecutionRequest<N: Network> {
+    transaction: Transaction<N>,
+    address: Address<N>,
+    program_id: ProgramID<N>,
+}
+
+impl<N: Network> ExecutionRequest<N> {
+    /// Sends the request to the given endpoint.
+    pub fn new(transaction: Transaction<N>, address: Address<N>, program_id: ProgramID<N>) -> Result<Self> {
+        ensure!(
+            matches!(transaction, Transaction::Execute(_, _, _)),
+            "Cannot create an execution request with a deploy transaction"
```

Comment:
```suggestion
            "Cannot create an execution request with a non-execution transaction"
```
^We intend to support other transaction types in the future, hence the greater specificity.